src/statistics.jl

```diff
@@ -1219,7 +1219,7 @@ function apply_statistic(stat::SmoothStatistic,
         aes.x = collect((x_min + nudge):((x_max - x_min) / num_steps):(x_max - nudge))
 
         if stat.method == :loess
-            aes.y = Loess.predict(loess(xs, ys, span=stat.smoothing), aes.x)
+            aes.y = Loess.predict(loess(xs, ys, span=stat.smoothing), Float64.(aes.x))
```

Comment:
When `aes.x` is a `DateTime` as in the linked issue, this will produce deprecation warnings on Julia 0.6. The proper way to convert a `DateTime` to a float is to call `Dates.value` and then convert.

One way to approach this would be to define some helper function that can handle dates appropriately:
```julia
@inline _aesconvert{T<:Real}(::Type{T}, d::DateTime) = convert(T, Dates.value(d))
@inline _aesconvert{T<:Real}(::Type{T}, x) = convert(T, x)
```
Then you can make this `_aesconvert.(Float64, aes.x)`.